File: barcosEnContienda/src/gestorAplicacion/Embarcaciones/Barco.java
package gestorAplicacion.Embarcaciones;

import gestorAplicacion.Personas.Capitanes;
import gestorAplicacion.Items.*;

public class Barco {
	public String nombre;
	public Capitanes capitan;
	public Faccion faccion;
	public int velocidad;
	public int ataque;
	public int defensa=5;
	public int deterioro=0;
	public Botin botin;
	public int nivelDeBusqueda=0;
	
	public Barco(String nombre, Faccion faccion, int velocidad, int ataque, Botin botin) {
		this.nombre = nombre;
		this.faccion=faccion;
		this.velocidad = velocidad;
		this.ataque = ataque;
		this.botin = botin;
	}

	public void atacar() {
	}
	
	public void robar() {
	}
	
	public void arribar() {
	}
	
	public void huir() {
	}
	
	public void vincularCapitan(Capitanes capitan) {
		this.capitan=capitan;
	}
	public String getNombre() {
		return nombre;
	}

	public void setNombre(String nombre) {
		this.nombre = nombre;
	}

	
	public Capitanes getCapitan() {
		return capitan;
	}

	public void setCapitan(Capitanes capitan) {
		this.capitan = capitan;
	}

	public int getVelocidad() {
		return velocidad;
	}

	public void setVelocidad(int velocidad) {
		this.velocidad = velocidad;
	}

	public int getAtaque() {
		return ataque;
	}

	public void setAtaque(int ataque) {
		this.ataque = ataque;
	}

	public int getDeterioro() {
		return deterioro;
	}

	public void setDeterioro(int deterioro) {
		this.deterioro = deterioro;
	}

	public Botin getBotin() {
		return botin;
	}

}

File: practica-2-grupo-1-equipo-7/src/gestorAplicacion/embarcaciones/barco.py
from gestorAplicacion.embarcaciones.faccion import Faccion
class Barco():
    _totalBarcos=0
    _totalBarcosDestruidos=0
    def __init__(self, nombre, facc, velocidad, ataque, botin):
        self._nombre=nombre
        self._velocidad=velocidad
        self._faccion=Faccion[facc]
        self._ataque=ataque
        self._botin=botin
        self._deterioro=0
        self._defensa=3
        self._nivel_de_busqueda=0
        
    
    def atacar(self,barco):
        barco.deterioro=barco.deterioro+self._ataque+self._capitan.liderazgo-barco._defensa
        
        if barco._deterioro<=0:
            barco._deterioro=0
        
        barco._capitan._tripulacion_cantidad_de_tripulantes-=1
        barco._velocidad=-1
        
        self._deterioro=self._deterioro+barco._ataque+barco._capitan._liderazgo-self._defensa
        
        if  self._deterioro<=0:
            self._deterioro=0
    
    def robar(self,barco):
        
        if barco._deterioro>50:
            self._botin._oro=self._botin._oro+barco._botin._oro
            self._botin._objetos.append(barco._botin._objetos)
        
            return True
        
        else:
            self._capitan._tripulacion._cantidad_de_tripulantes=int(self._capitan._tripulacion._cantidad_de_tripulantes-(barco._capitan._liderazgo*0.2)-1)
    
    def robarIsla(self,isla):
        print('Asalto Isla')
        if self._deterioro<=50:
            self._capitan._tripulacion._cantidad_de_tripulantes=int(self._capitan._tripulacion._cantidad_de_tripulantes-(self._capitan._liderazgo*0.2)-1)
            isla._tripulacion._cantidad_de_tripulantes=int(isla._tripulacion._cantidad_de_tripulantes-(self._capitan._liderazgo*0.2)-3)
            
            if self._capitan._tripulacion._cantidad_de_tripulantes>0:
                self._botin._oro = self._botin._oro + isla._botin._oro
                isla._botin._oro = 0
                self._botin._objetos.append(isla._botin._objetos.get(0))
				#isla._botin._objetos.clear()
                return True

            else:
                self._capitan._tripulacion._cantidad_de_tripulantes = 0
                return False

        else:
            self._capitan._tripulacion._cantidad_de_tripulantes=int(self._capitan._tripulacion._cantidad_de_tripulantes-(self._capitan._liderazgo*0.2)-3)
            isla._tripulacion._cantidad_de_tripulantes=int(isla._tripulacion.cantidad_de_tripulantes-(self._capitan.liderazgo*0.2)-1)
            if self._capitan._tripulacion.cantidad_de_tripulantes>0:
                self._botin._oro = self._botin._oro + isla._botin._oro
                isla.botin.oro = 0
                self._botin._objetos.append(isla._botin._objetos.get(0))
				#isla._botin._objetos.clear()
                return True
            else:
                self._capitan._tripulacion._cantidad_de_tripulantes = 0
                return False
			
    def setDeterioro(self,deterioro):
        self._deterioro = deterioro

    def vincularCapitan(self,capitan):
        self._capitan=capitan